basic/http/collect-user.js

```diff
@@ -25,13 +28,49 @@ class User {
 };
 
 const user = new User();
+
 router.get( '/users', async ctx => {
-  ctx.body = {data:user.data.map((e)=>{
+  ctx.body = {
+    data:user.data.map((e)=>{
       return {id:e.id,name:e.name}; 
     })
   };
+});
 
+router.post('/users',async ctx => {
+  
+  user.data.push(ctx.request.body);
+  
+  ctx.body = {
+    data:user.data.map((e)=>{
+      return {id:e.id,name:e.name}; 
+    })
+  };
 });
+
+router.patch('/users/:id',async ctx => {  
+
+  const updateUser = user.data;
+  for (const obj of updateUser) {
+    if (obj.updateUser.id === ctx.params.id) {
```

Comment:
`obj.updateUser.id` is undefined here. Where did you get `obj.updateUser`? 
This is the usage of `for..of`
```ts
const data = [{name: 'Jack'}, {name: 'Jane'}];
for (const obj of data) {
  console.log(obj); 
  /**
   * 0 - { name: 'Jack' }
   * 1 - { name: 'Jane' }
   * The iteration is an object.
   */
}
```

So in your codebase where did you get the `updateUser`?. In the first, place `obj` only contains, `id`, `name`, and `password`.